elastic/apm-aws-lambda
# Make Secret token / API Key optional

Resolves #180 

### Changes
If none of `ELASTIC_APM_SECRET_TOKEN` or `ELASTIC_APM_API_KEY` are detected, the extension logs a warning instead of exiting. The docs have been updated accordingly. Note that the corresponding agent still needs to be added to the list of services allowed to communicate with the Server / Integration anonymously.

### How to test
- [Setup](https://www.elastic.co/guide/en/apm/agent/python/current/lambda-support.html) a Python Lambda function with the Python APM Agent and the Lambda extension.
- Add the `python` service to the list of services allowed to communicate anonymously with your Server / Integration
- In the Lambda configuration, only set the following environment variables:
```
ELASTIC_APM_LAMBDA_APM_SERVER=<Your APM Server>
ELASTIC_APM_LOG_LEVEL=debug 
```
The `debug` log level allows us to check that the APM Server returns a 202 response.
